### PR TITLE
Login/logout capability

### PR DESCRIPTION
**Pull request by br0ziliy**
Added ability to "logout" off the application.
Right now this does not revoke the token itself in the Vault because I found
that this way you're able to revoke the root token (at least in Vault dev mode).

Also added some token info to the screen.

### CODE REVIEW

**Comment by br0ziliy** [comment]
I'm not sure about the change on line 130 in vault.js though, but it looks logical to me.

**Comment by br0ziliy** [comment]
This will help when I add multiple auth backends support (GitHub, userpass and LDAP are on the way).

File: src/vault.js [hunk 1]
```diff
@@ -170,6 +179,27 @@ var Page = function() {
...
+    });
+  }
+
+  self.getTokenInfo = function() {
+    if (self.token()) self.apiToken();
+    if (self.vaultTokenResponse()) {
+        var t = JSON.parse(self.vaultTokenResponse());
+        var id = t.id;
+        var name = t.display_name;
+        return name + '@' + id;
```
Comment (AMeng) [inline src/vault.js]: I don't think we want to display the token in the main UI. You can click on the `System` dropdown and select `Token Information` to see the token and its details. How about just the name here instead?

image

**Comment by AMeng** [comment]
Thanks for the PR! Would you mind adding a `Login` button on the form? Having the login attempt happen when the token field changes is a bit strange if the form is going to disappear.

**Comment by br0ziliy** [comment]
Thanks for comments.
Main idea behind this change was to simplify the process of identification of credentials that the app is currently using. `Token Information` pop-up is useful, but hard to read, and you have to click around to get to the window.
I agree though that the current implementation is a bit ugly :)
So, which one of those you'd prefer?
2016-08-16-181315_205x170_scrot
2016-08-16-181217_343x201_scrot
2016-08-16-181135_398x175_scrot

The login button sounds like a good idea, too, will add that.
And I will put `Logout` like this, if you don't mind:
2016-08-16-182006_266x217_scrot

Still not clear how to identify the Vault server one is connected to... Maybe move that data to the `#vaultHealthModal`?

**Comment by AMeng** [comment]
> So, which one of those you'd prefer?

I would say the first one, just because it keeps the token hidden until the user explicitly asks to view it.

> And I will put Logout like this, if you don't mind

That looks great!

**Comment by br0ziliy** [comment]
@AMeng pushed the updates as discussed.

**Comment by AMeng** [comment]
Thanks for the great work!

**Pull request merged**